src/explore-education-statistics-common/src/modules/find-statistics/hooks/useTableQuery.ts

```diff
@@ -6,15 +6,19 @@ import tableBuilderService from '@common/services/tableBuilderService';
 export default function useTableQuery(
   releaseId: string,
   dataBlockParentId: string,
-  boundaryLevelId?: number,
+  boundaryLevel?: number
 ): AsyncRetryState<FullTable> {
-  return useAsyncRetry(async () => {
-    const response = await tableBuilderService.getDataBlockTableData(
-      releaseId,
-      dataBlockParentId,
-      boundaryLevelId,
-    );
+  return  useAsyncRetry(async () => {
+    const [tableData, locationGeoJson] = await Promise.all([
+      tableBuilderService.getDataBlockTableData(
+        releaseId,
+        dataBlockParentId,
+      ),
+      boundaryLevel !== undefined ? tableBuilderService.getLocationGeoJson(releaseId, dataBlockParentId, boundaryLevel): undefined
+    ])
 
-    return mapFullTable(response);
+    
+
+    return mapFullTable({...tableData, subjectMeta: {...tableData.subjectMeta, locations: locationGeoJson ?? tableData.subjectMeta.locations}});
```

Comment:
Yo, should I also bring in getKeyStat (see **useKeyStateQuery.ts**) into **tableBuildQueries**? Looks like currently `useKeyStatQuery` has a setter, which never seems to be used so I think this can be replaced with a `useQuery` implementation. The reason I ask is because now that I've removed (locally) the old `useTableQuery` this key stat query now needs refactoring.

Reply:
Yep that's definitely one option.

Alternatively, it's fine to also keep `useKeyStatQuery` and refactor it. Think it'd be fairly straightforward via the `select` option to do the transform e.g.

```ts
export default function useKeyStatQuery(...): UseQueryResult<KeyStatResult>
  return useQuery({
    ...tableBuilderQueries.getDataBlockTable(...),
    select: data => {
      // Can transform the data in some way here
      return data;
    }
  });
}
```

I don't massively mind which option we go for tbh, so will leave to your discretion.

Btw, it's also worth giving this article on [data transformations](https://tkdodo.eu/blog/react-query-data-transformations) a read. It's by one of the lead maintainers for React Query, so it's got some interesting things you'll want to absorb.